Tasks/Maven/maven.ps1

```diff
@@ -28,14 +32,37 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
+# If JAVA_HOME is being set by choosing a JDK version find the path to that specified version else use the path given by the user
+$jdkPath = $null
+if($javaHomeSelection -eq 'JDKVersion')
 {
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
+    Write-Verbose "Using JDK version to find and set JAVA_HOME"
+    # If the JDK version is not the deafult set the jdkPath to the new JDK version selected
+    if($jdkVersion -and $jdkVersion -ne "default")
     {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
+        $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
+        if (!$jdkPath) 
+        {
+            throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
+        }
     }
+}
+else
+{
+    Write-Verbose "Using path from user input to set JAVA_HOME"
+    if($jdkUserInputPath -and (Test-Path -Path $jdkUserInputPath))
+    {
+        $jdkPath = $jdkUserInputPath
+    }
+    else
+    {
+         throw "Path given for the JDK does not exist. Please make sure to enter in a valid path."
```

Comment:
We could make this message more succinct with:
"The specified JDK path does not exist. Please provide a valid path."